### ISSUE TITLE
Connection reset by peer when using aws s3 cp

### ISSUE THREAD

**Issue by sanalbert** [aws/aws-cli#6876]
### Describe the bug

A connection reset error in copying a file from a public S3 bucket to a local directory.

### Expected Behavior

I am trying to download a large(1.7TB) file from the [NIH NCBI Sequence Read Archive (SRA) on AWS](https://registry.opendata.aws/ncbi-sra/) to my local directory. It is a simple `aws s3 cp` job that should create a copy of the file within the designated directory.

### Current Behavior

The download runs with a speed of ~8MB/s for about 12 to 18 hours and after that is terminated with the following message:

> download failed: s3://sra-pub-src-1/SRZ189891/722g.990.SNP.INDEL.chrAll.vcf.1 to ./722g.vcf.1 ("Connection broken: ConnectionResetError(104, 'Connection reset by peer')", ConnectionResetError(104, 'Connection reset by peer'))

Nothing that looks like an error message in the --debug, just the last line printing the connection error.

> $ tail debug
> 2022-04-08 23:23:17,999 - ThreadPoolExecutor-0_0 - s3transfer.utils - DEBUG - Acquiring 0
> 2022-04-08 23:23:17,999 - ThreadPoolExecutor-0_0 - s3transfer.utils - DEBUG - Releasing acquire 0/None
> 2022-04-08 23:23:17,999 - ThreadPoolExecutor-2_0 - s3transfer.tasks - DEBUG - IORenameFileTask(transfer_id=0, {'final_filename': '/storage/SRC/alix/722g.vcf.1'}) about to wait for the following futures []
> 2022-04-08 23:23:17,999 - ThreadPoolExecutor-2_0 - s3transfer.tasks - DEBUG - IORenameFileTask(transfer_id=0, {'final_filename': '/storage/SRC/alix/722g.vcf.1'}) done waiting for dependent futures
> 2022-04-08 23:23:32,376 - ThreadPoolExecutor-2_0 - s3transfer.utils - DEBUG - Releasing acquire 0/None
> download failed: s3://sra-pub-src-1/SRZ189891/722g.990.SNP.INDEL.chrAll.vcf.1 to ./722g.vcf.1 ("Connection broken: ConnectionResetError(104, 'Connection reset by peer')", ConnectionResetError(104, 'Connection reset by peer'))
> 2022-04-08 23:23:32,378 - Thread-1 - awscli.customizations.s3.results - DEBUG - Shutdown request received in result processing thread, shutting down result thread.

the `--debug` log file is really large and my text editor is having problems handling it; I will try to add it asap.

### Reproduction Steps

I am using a simple copy command:

`aws s3 cp --no-sign-request s3://sra-pub-src-1/SRZ189891/722g.990.SNP.INDEL.chrAll.vcf.1 ./722g.vcf.1`

### Possible Solution

Maybe if there is a way to speed up the transfer, the download could finish before the error occurs? I've tried configuring the preferred_transfer_client to crt but there were no significant improvements.

### CLI version used

aws-cli/2.5.2 Python/3.9.11 Linux/3.10.0-1160.31.1.el7.x86_64 exe/x86_64.centos.7 prompt/off

### Environment details (OS name and version, etc.)

CentOS Linux 7 (Core)

**Comment by tim-finnigan** [comment]
Thanks @sanalbert for reaching out. Were you able to get the full debug logs? Those would help us investigate this issue further.

Also ~8MB/s download speed seems slow. Have you tried running the command on another machine or network connection?

**Comment by sanalbert** [comment]
I've loaded the full debug log file [here](https://github.com/sanalbert/sanalbert/blob/07bea47aba0cae60005ad3fd811fd9f565187472/Desktop/debuglog.txt.zip)

Thanks @tim-finnigan, and about the speed we've run the command on a cluster server(Linux) and an iMac(tried both MacOS and Linux versions) but had the same results.

**Comment by tim-finnigan** [comment]
Hi @sanalbert thanks for following up. The `ConnectionResetError` was also mentioned in https://github.com/boto/boto3/issues/1072 where someone commented:


> This error is originating from the network, somewhere in the network path something is resetting the TCP connection and causing that error. Since this only occurs on your server, it's likely something amiss with the network that the server itself is connected to.


(The `ConnectionResetError` was mentioned in issues https://github.com/boto/boto3/issues/2495 and https://github.com/aws/aws-cli/issues/5296 as well)

Maybe you could try setting [tcp_keepalive](https://docs.aws.amazon.com/cli/latest/topic/config-vars.html) to true in your config file?

**Comment by sanalbert** [comment]
@tim-finnigan Thank you for the information, we are trying again with `tcp_keepalive = true`
The command is running for 22 hrs now. We'll follow up when we get (hopefully succesful) results.
Apparantly the download speed was at least partly due to our CPU; recently we cleaned the CPU load and this time we reached 14MiB/s. Also I have heard that our institute network might have some constraints related to download speed.
